go/api/list_commands.go

```diff
@@ -0,0 +1,206 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// Supports commands and transactions for the "List Commands" group for standalone and cluster clients.
+//
+// See [valkey.io] for details.
+//
+// [valkey.io]: https://valkey.io/commands/?group=list
+type ListCommands interface {
+	// Inserts all the specified values at the head of the list stored at key. elements are inserted one after the other to the
+	// head of the list, from the leftmost element to the rightmost element. If key does not exist, it is created as an empty
+	// list before performing the push operation.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key      - The key of the list.
+	//  elements - The elements to insert at the head of the list stored at key.
+	//
+	// Return value:
+	//  A api.Result[int64] containing the length of the list after the push operation.
+	//
+	// For example:
+	//  result, err := client.LPush("my_list", []string{"value1", "value2"})
+	//  result.Value(): 2
+	//  result.IsNil(): false
+	//
+	// [valkey.io]: https://valkey.io/commands/lpush/
+	LPush(key string, elements []string) (Result[int64], error)
+
+	// Removes and returns the first elements of the list stored at key. The command pops a single element from the beginning
+	// of the list.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key - The key of the list.
+	//
+	// Return value:
+	//  The Result[string] containing the value of the first element.
+	//  If key does not exist, [api.CreateNilStringResult()] will be returned.
```

Comment:
```suggestion
	//  If key does not exist, `nil` will be returned.
```
For the end user it does not matter where null comes from. We should describe what returned and when.

Reply:
Here, the `api.CreateNilStringResult()` is returning `nil` object, whereas it's returning `api.Result[string]{val: "", isNil: true}` object. To not make it too verbose, we decided to replace it with a function and link in the docs. 

Please let me know if you think otherwise.